rct229/schema/validate.py

```diff
@@ -21,7 +21,112 @@
+    referenced_id_list = find_all(
+        "$.buildings[*].elevators[*].motor_location_zone", rmd
+    )
+    referenced_id_list.extend(
+        find_all("$.buildings[*].elevators[*].cab_location_zone", rmd)
+    )
+    referenced_id_list.extend(
+        find_all("$.buildings[*].refrigerated_cases[*].zone", rmd)
+    )
+    referenced_id_list.extend(
+        find_all("$.service_water_heating_equipment[*].compressor_zone", rmd)
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.service_water_heating_equipment[*].compressor_heat_rejection_zone", rmd
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].zones[*].zonal_exhaust_fan.motor_location_zone",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].zones[*].terminals[*].fan.motor_location_zone",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].fan_system.supply_fans[*].motor_location_zone",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].fan_system.return_fans[*].motor_location_zone",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].fan_system.relief_fans[*].motor_location_zone",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].fan_system.exhaust_fans[*].motor_location_zone",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all("$.service_water_heating_equipment[*].tank.location_zone", rmd)
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.service_water_heating_equipment[*].solar_thermal_systems[*].tank.location_zone",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.service_water_heating_distribution_systems[*].tanks[*].location_zone",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].zones[*].surfaces[*].adjacent_zone",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].zones[*].transfer_airflow_source_zone",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.service_water_heating_distribution_systems[*].service_water_piping[*].location_zone",
+            rmd,
+        )
+    )
```

Comment:
I wonder if we can create a utility function called:
`find_all_by_jsonpaths` which takes in a list of jsonpaths and return a list of matching results. Will that be too confusing?

Reply:
Also, please run black format before merging.

Reply:
The `find_all_by_jsonpaths` utility makes sense to me. 

The black format test is failing from Rule 4-2, I don't believe it is originating from the code updated in this PR.